---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r setup, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

# ezcor
A tool to perform correlation in a batch mode and return tidy result

## :star: Motivation
**Correlation analysis** is very common, here we wrap several functions for correlation analysis to help us analyze datasets such as **TCGA** (The Cancer Genome Atlas) dataset in bioinformatics

## :arrow_double_down: Installation
```{r, eval=FALSE}
if(!require("remotes")){install.packages("remotes")}
remotes::install_github("Byronxy/ezcor")
```

## :rocket: Support functions

```{r, eval=FALSE}
ezcor() ##run basic correlation
ezcor_batch() ##run basic correlation in a batch mode
ezcor_bicor() ##run WCGAA Biweight Midcorrelation correlation
ezcor_bicor_batch() ##run WCGAA Biweight Midcorrelation correlation in a batch mode
ezcor_partial_cor() ##run partial correlation
ezcor_partial_cor_batch() ##run partial correlation in a batch mode
```

## :sunny: Examples
We select HALLMARK_HYPOXIA pathway genes from TCGA pan-cancer dataset to illustrate examples
```{r}
data("exprSet", package = "ezcor", envir = environment())
exprSet[1:5,1:5]
g1 <- colnames(exprSet)[3]
g2 <- colnames(exprSet)[4]
res <- ezcor::ezcor(data= exprSet,
                      split = TRUE,
                      split_var = "tissue",
                      var1 = g1,
                      var2 = g2,
                      cor_method = "spearman",
                      adjust_method = "none",
                      sig_label = TRUE,
                      verbose = TRUE)
head(res)
```

## :writing_hand: Author
* [Yi Xiong](https://github.com/Byronxy)
* [Shixiang Wang](https://github.com/ShixiangWang)


